awx/main/dispatch/periodic.py

```diff
@@ -0,0 +1,37 @@
+import threading
+import time
+
+from django.conf import settings
+from django.db import connections
+from schedule import Scheduler
+
+from awx.main.dispatch.worker import TaskWorker
+
+
+class Scheduler(Scheduler):
+
+    def run_continuously(self):
+        cease_continuous_run = threading.Event()
+
+        class ScheduleThread(threading.Thread):
+            @classmethod
+            def run(cls):
+                while not cease_continuous_run.is_set():
+                    for conn in connections.all():
+                        # If the database connection has a hiccup, re-establish a new
+                        # connection
+                        conn.close_if_unusable_or_obsolete()
```

Comment:
interesting to pick this back up, that's probably smart

Reply:
Yea, these processes are long-lived, and I remember that `django-celery` basically did this after every new message it received just to makes sure database hiccups didn't cause issues.